docs/style-guidelines.md

```diff
@@ -1,20 +1,313 @@
+      * tabs saved as spaces, `1 tab` = `4 spaces` (characters)
+    * only one space between code elements is allowed
+    * space after a comma between function arguments - 🟢 `CollectItem(myObject, 0, 1);` 🔴 `CollectItem(myObject,0,1);`
+    * space before flow control conditions - 🟢 `while (x < y)` 🔴 `while(x < y)`
+    * space before and after comparison operators - 🟢 `if (x == y)` 🔴 `if (x==y)`
+  * **Vertical spaces**
+    * one blank line is used for vertical separation and grouping
+    * one blank line is always used to separate groups (field, properties, method definitions)
+    * 🔴 two and more blank lines successively are not allowed
+  * **Line breaks and curly braces** `{}`
+    * Curly brace is always placed on a new line (Allman/BSD style):
+      * line break before the opening brace `{`
+      * line break between the closing brace `}` and `else`
+    * avoid brackets when the body is a one line
+    * put the body on the new line in most cases 
+    * it is allowed to put the body in the same line when it contains only one simple interruption (like `return`/ `break`/ `continue`)
+```csharp
+if(!IsInRange()) return;    // interruptions are in the same line
+
+if(IsInRange())             // body on the new line; no brackets
+    Fire()
+else
+{
+    CalculateDistance();
+    MoveToEnemy();
+}
+```
+* **Expression-bodies**
+  * remove brackets in most cases where it is possible (for loops, `if`-`else`, methods, and properties)
+    * ⚠️exception: always use brackets for **Unity callbacks** 
+  * **Properties** - placed on the same line 
+  * **Methods** - placed on the new line
+```csharp
+public bool IsInitialized => isInitialized;    // Property - body placed on the same line =>
+
+private void Awake()                           // Unity-callback - use brackets even when expression-body is possible  
+{
+    var collider = GetComponent<Collider>;
+}
+
+public void Initialize() =>                    // Methods - body placed on the new line after =>
+    SubrcribeToEvents();
+
+private void TrimAll()
+{
+    foreach (string device in InputDevices)    // Braces removed
+        device.Trim();
+}
+```
+
+### Attribute usages
+* **Class**
+  * Use `[RequireComponent(typeof(MyComponent))]` attribute when you cache components via `GetComponents` on `Awake` and `Start` callbacks
+  * Use `[DisallowMultipleComponent]` attribute when there should be only one `Component` of such type on the `GameObject`
+* **Methods**
+  * Use `[PublicAPI]` attribute if `public method` is exposed to be called from **outside** of the solution 
+  * Use `[UsedImplicitly]` attribute for implicitly called methods, such as calls from Unity animation events, via Unity `GameObject.SendMessage()`, `GameObject.BroadcastMessage()` and similar. 
+  * Use `[Button]` attribute from `EasyButtons` (requires reference in `.asmdef` file) instead of `[ContectMenu]` attribute. Consider using `Mode` parameter of this attribute for `Editor`/`PlayMode`-only methods.
```

Comment:
```suggestion
  * Use `[Button]` attribute from `EasyButtons` (requires reference in `.asmdef` file) instead of `[ContextMenu]` attribute. Consider using `Mode` parameter of this attribute for `Editor`/`PlayMode`-only methods.
```